day1/es1/SwitchButton.qml

```diff
@@ -0,0 +1,64 @@
+import QtQuick 2.0
+
+Row {
+    id: switchButton
+
+    width: 160
+    height: 40
+    anchors.centerIn: parent
+    state: "unchecked"
+
+    LocalButton {
+        stateName: "checked"
+        colorForTheStateChecked: "blue"
+        colorForTheStateUnchecked: "lightgray"
+        text: qsTr("on")
+    }
+    LocalButton {
+        stateName: "unchecked"
+        colorForTheStateChecked: "lightgray"
+        colorForTheStateUnchecked: "Darkgray"
+        text: qsTr("off")
+    }
+
+    states: [
+        State {
+            name: "checked"
+        },
+        State {
+            name: "unchecked"
+        }
+    ]
+
+    component LocalButton: Rectangle {
```

Comment:
Qui vedo che hai usato una componente inline.
Solitamente le componenti inline si usano molto di rado: questo ovviamente è un esercizio molto contenuto, ma se immagini di essere in un progetto grande è verosimile lche vengano usate molte componenti diverse per comporne di nuove, e averle tutte inline significherebbe:
- avere file QML molto lunghi
- avere difficoltà nel riutilizzo, essendo praticamente costretti a fare copia-incolla delle varie componenti inline dentro altri file che le vogliono a loro volta usare

Quello che si farebbe solitamente in questo caso, è creare un nuovo file LocalButton.qml (allo stesso livello di directory di SwitchButton.qml) e definire la componente lì dentro.
Collateralmente a questo, dovrebbe anche risolverti il problema di syntax highlighting.

Che ne dici? Ti va di provare?

Ricordati che per lavorare su questo devi trovarti sul branch `day1_es1`:
```
git checkout day1_es1
```

Quando hai finito, fai `git add` sui file che hai modificato, prepara un commit con `git commit -m "descrizione"` e carica qui le tue modifiche con `git push`.

Se qualcosa non ti è chiaro, puoi pingarmi sul nostro canale slack dedicato! :)

Reply:
Fatto, passato il componente in un altro file :+1: